NitroxClient/GameLogic/Spawning/Metadata/Processor/IOxygenSourceMetadataProcessor.cs

```diff
@@ -0,0 +1,22 @@
+﻿using NitroxClient.GameLogic.Spawning.Metadata.Processor.Abstract;
+using NitroxModel.DataStructures.GameLogic.Entities.Metadata;
+using UnityEngine;
+
+namespace NitroxClient.GameLogic.Spawning.Metadata.Processor;
+
+public class IOxygenSourceMetadataProcessor : EntityMetadataProcessor<IOxygenSourceMetadata>
+{
+    public override void ProcessMetadata(GameObject gameObject, IOxygenSourceMetadata metadata)
+    {
+        IOxygenSource iOxygenSource = gameObject.GetComponent<IOxygenSource>();
+
+        if (iOxygenSource != null)
+        {
+            iOxygenSource.AddOxygen(metadata.AvailableOxygen);
```

Comment:
As this can also possibly be executed on existing tanks I would first calculate the delta of `GetOxygenAvailable()` and `metadata.AvailableOxygen`, to then use it for `AddOxygen()` or `RemoveOxygen()` depending on plus or minus.